rules/0131.go

```diff
@@ -0,0 +1,202 @@
+							methodName+"Request",
+						),
+						Suggestion: methodName + "Request",
+						Descriptor: m,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// The Get standard method should only have expected fields.
+func checkGetRequestMessageNameField() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0131", "request_message", "name_field"),
+			Description:  "check that a name field is present",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get methods for the purpose of this rule;
+				// ignore everything else.
+				if !isGetMethod(m) {
+					return
+				}
+
+				// Rule check: Establish that a name field is present.
+				nameField := m.Input().Fields().ByName("name")
+				if nameField == nil {
+					problems = append(problems, lint.Problem{
+						Message:    fmt.Sprintf("method %q has no `name` field", m.Name()),
+						Descriptor: m.Input(),
+					})
+					return problems, nil
+				}
+
+				// Rule check: Establish that the name field is a string.
+				if nameField.Kind() != p.StringKind {
+					problems = append(problems, lint.Problem{
+						Message:    "`name` field on Get RPCs should be a string",
+						Descriptor: nameField,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Get methods should not have unrecognized fields.
+func checkGetRequestMessageUnknownFields() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0131", "request_message", "unknown_fields"),
+			Description:  "check that there are no unknown fields",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get methods for the purpose of this rule;
+				// ignore everything else.
+				if !isGetMethod(m) {
+					return
+				}
+
+				// Rule check: Establish that there are no unexpected fields.
+				allowedFields := map[string]struct{}{
+					"name":      struct{}{}, // AIP-131
+					"read_mask": struct{}{}, // AIP-158
+					"view":      struct{}{}, // AIP-158
+				}
+				fields := m.Input().Fields()
+				for i := 0; i < fields.Len(); i++ {
+					field := fields.Get(i)
+					if _, ok := allowedFields[string(field.Name())]; !ok {
+						problems = append(problems, lint.Problem{
+							Message: fmt.Sprintf(
+								"Get RPCs should only only contain fields explicitly described in AIPs, not %q.",
+								string(field.Name()),
+							),
+							Descriptor: field,
```

Comment:
Same question -- where should the lint warning show up? In the service method section, or the message field?

Reply:
I want to to show up on the field in this case.